src/NServiceBus.RabbitMQ/Routing/ConventionalRoutingTopology.cs

```diff
@@ -76,6 +77,26 @@ public void Initialize(IModel channel, string mainQueue)
             channel.QueueBind(mainQueue, mainQueue, string.Empty);
         }
 
+        public string SetupDelay(IModel channel, long delay)
+        {
+            channel.ExchangeDeclare("delay-triggered", "fanout", useDurableExchanges);
+            channel.QueueDeclare("delay-triggered", useDurableExchanges, false, false, null);
+            channel.QueueBind("delay-triggered", "delay-triggered", "");
+
+            var address = $"delay-{delay}";
+
+            channel.ExchangeDeclare(address, "fanout", useDurableExchanges);
+
+            var arguments = new Dictionary<string, object>();
+            arguments.Add("x-message-ttl", delay);
+            arguments.Add("x-dead-letter-exchange", "delay-triggered");
```

Comment:
So basically we create a queue, we set TTL and allow the messages to be discarded, as dead letter queue setting the delay-triggered queue from which we pick messages and send to its original destination. 

That is crafty @bording  :clap: